code/game/machinery/suit_storage_unit.dm

```diff
@@ -1,184 +1,406 @@
+	cut_overlays()
 
+	if(uv)
+		if(uv_super)
+			add_overlay("super")
+		else if(occupant)
+			add_overlay("uvhuman")
+		else
+			add_overlay("uv")
+	else if(state_open)
+		if(stat & BROKEN)
+			add_overlay("broken")
+		else
+			add_overlay("open")
+			if(suit)
+				add_overlay("suit")
+			if(helmet)
+				add_overlay("helm")
+			if(storage)
+				add_overlay("storage")
+	else if(occupant)
+		add_overlay("human")
+
+/obj/machinery/suit_storage_unit/attackby(obj/item/I as obj, mob/user as mob, params)
+	if(!is_operational())
+		return
+	if(isscrewdriver(I))
+		panel_open = !panel_open
+		playsound(loc, I.usesound, 100, 1)
+		to_chat(user, text("<span class='notice'>You [panel_open ? "open up" : "close"] the unit's maintenance panel.</span>"))
+		updateUsrDialog()
 
 /obj/machinery/suit_storage_unit/power_change()
-	if( powered() )
-		src.ispowered = 1
-		stat &= ~NOPOWER
-		src.update_icon()
+	..()
+	if(!is_operational() && state_open)
+		open_machine()
+		dump_contents()
+	update_icon()
+
+
+/obj/machinery/suit_storage_unit/proc/dump_contents()
+	dropContents()
+	helmet = null
+	suit = null
+	mask = null
+	storage = null
+	occupant = null
+
+/obj/machinery/suit_storage_unit/deconstruct(disassembled = TRUE)
+	open_machine()
+	dump_contents()
+	new /obj/item/stack/sheet/metal (loc, 2)
+	qdel(src)
+
+/obj/machinery/suit_storage_unit/MouseDrop_T(atom/A, mob/user)
+	if(user.stat || user.lying || !Adjacent(user) || !Adjacent(A) || !isliving(A))
+		return
+	var/mob/living/target = A
+	if(!state_open)
+		to_chat(user, "<span class='warning'>The unit's doors are shut!</span>")
+		return
+	if(!is_operational())
+		to_chat(user, "<span class='warning'>The unit is not operational!</span>")
+		return
+	if(occupant || helmet || suit || storage)
+		to_chat(user, "<span class='warning'>It's too cluttered inside to fit in!</span>")
+		return
+
+	if(target == user)
+		user.visible_message("<span class='warning'>[user] starts squeezing into [src]!</span>", "<span class='notice'>You start working your way into [src]...</span>")
 	else
-		spawn(rand(0, 15))
-			src.ispowered = 0
-			stat |= NOPOWER
-			src.islocked = 0
-			src.isopen = 1
-			src.dump_everything()
-			src.update_icon()
-
-
-/obj/machinery/suit_storage_unit/ex_act(severity)
-	switch(severity)
-		if(1.0)
-			if(prob(50))
-				src.dump_everything() //So suits dont survive all the time
-			qdel(src)
-			return
-		if(2.0)
-			if(prob(50))
-				src.dump_everything()
-				qdel(src)
+		target.visible_message("<span class='warning'>[user] starts shoving [target] into [src]!</span>", "<span class='userdanger'>[user] starts shoving you into [src]!</span>")
+
+	if(do_mob(user, target, 30))
+		if(occupant || helmet || suit || storage)
 			return
+		if(target == user)
+			user.visible_message("<span class='warning'>[user] slips into [src] and closes the door behind [user.p_them()]!</span>", "<span class=notice'>You slip into [src]'s cramped space and shut its door.</span>")

```

Comment:
need apostrophes for `span class=notice`

Reply:
This is NOT resolved, why'd you resolve this?

Reply:
@variableundefined ...... it IS resolved.... are you looking at the latest version of the code? This was fixed in 6b8c598.... two days ago. https://github.com/ParadiseSS13/Paradise/pull/9810/commits/6b8c5983fd4d2e20b2781f8c2ed48b47e25b3c97